lightning/src/chain/chainmonitor.rs

```diff
@@ -556,6 +556,25 @@ where C::Target: chain::Filter,
 	}
 }
 
+impl<ChannelSigner: WriteableEcdsaChannelSigner, C: Deref, T: Deref, F: Deref, L: Deref, P: Deref> ChainMonitor<ChannelSigner, C, T, F, L, P>
+	where C::Target: chain::Filter,
+	      T::Target: BroadcasterInterface,
+	      F::Target: FeeEstimator,
+	      L::Target: Logger,
+	      P::Target: Persist<ChannelSigner>,
+	      ChannelSigner: Clone
+{
+	/// Gets all the funding outpoints with each [`ChannelMonitor`] being monitored.
+	///
+	/// Note that [`ChannelMonitor`]s are not removed when a channel is closed as they are always
+	/// monitoring for on-chain state resolutions.
+	pub fn get_monitors(&self) -> HashMap<OutPoint, ChannelMonitor<ChannelSigner>> {
+		self.monitors.read().unwrap().iter().map(|(outpoint, holder)| {
```

Comment:
there's already a `list_monitors` that gives a `Vec<Outpoint>` and `get_mointor(outpoint)` but these require doing a bunch of locks. this replaces so we can get them all in one lock. The idea for this is to be able to get all of our monitors so we can create a backup without having to iterate over each one ourself.

Doing an iterator might be better than the hashmap here

Reply:
I don't think this can be an iterator because of the RwLock, maybe I'm doing something wrong but always getting `returns a value referencing data owned by the current function` that seems to only be solved by doing an allocation